src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentchecker/DuplicateFilenameDocumentCheckerTest.java

```diff
@@ -0,0 +1,972 @@
+            Arguments.of(FinremCaseData.builder()
+                .outOfFamilyCourtResolution(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .additionalListOfHearingDocuments(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .variationOrderDocument(DUPLICATED_CASE_DOCUMENT)
+                .build()),
+            Arguments.of(FinremCaseData.builder()
+                .consentVariationOrderDocument(DUPLICATED_CASE_DOCUMENT)
+                .build())
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void testGetWarningsOnExistingCase(FinremCaseData caseData)
+        throws DocumentContentCheckerException {
+        List<String> warnings = underTest.getWarnings(DUPLICATED_CASE_DOCUMENT, new byte[0],
+            FinremCaseDetails.builder()
+                .data(caseData)
+                .build(),
+            FinremCaseDetails.builder().build());
+        assertDuplicateFilenameWarning(warnings);
+    }
+
+    @Test
+    void testGetWarningsOnUploadingDuplicatedFilesAtTheSameTimeConsented() throws DocumentContentCheckerException {
+        List<String> warnings = underTest.getWarnings(DUPLICATED_CASE_DOCUMENT, new byte[0],
+            FinremCaseDetailsBuilderFactory.from(CaseType.CONSENTED).build(),
+            FinremCaseDetailsBuilderFactory.from(CaseType.CONSENTED, FinremCaseData.builder()
+                .uploadDocuments(List.of(
+                    UploadDocumentCollection.builder().value(UploadDocument.builder()
+                            .documentComment("1")
+                            .documentLink(DUPLICATED_CASE_DOCUMENT)
+                            .build())
+                        .build(),
+                    UploadDocumentCollection.builder().value(UploadDocument.builder()
+                            .documentComment("2")
+                            .documentLink(DUPLICATED_CASE_DOCUMENT)
+                            .build())
+                        .build()
+                ))).build());
+        assertDuplicateFilenameWarning(warnings);
+    }
+
+    @Test
+    void shouldNotGetWarningsOnUploadingDuplicatedFilesAtTheSameTimeConsented() throws DocumentContentCheckerException {
+        List<String> warnings = underTest.getWarnings(DUPLICATED_CASE_DOCUMENT, new byte[0],
+            FinremCaseDetailsBuilderFactory.from(CaseType.CONSENTED).build(),
+            FinremCaseDetailsBuilderFactory.from(CaseType.CONSENTED, FinremCaseData.builder()
+                .uploadDocuments(List.of(
+                    UploadDocumentCollection.builder().value(UploadDocument.builder()
+                            .documentComment("1")
+                            .documentLink(DUPLICATED_CASE_DOCUMENT)
+                            .build())
+                        .build()
+                ))).build());
+
+        assertThat(warnings).isEmpty();
+    }
+
+    @Test
```

Comment:
testGetWarningsOnUploadingDuplicatedFilesAtTheSameTimeContented. Probably mean Contested (not contented)

Reply:
sorry for the typo